runtime/onert/backend/train/ops/BinaryArithmeticLayer.cc

```diff
@@ -0,0 +1,96 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "BinaryArithmeticLayer.h"
+
+#include "OperationUtils.h"
+
+#include <cker/Shape.h>
+#include <cker/train/operation/BinaryArithmetic.h>
+#include <cker/operation/BinaryArithmeticOps.h>
+#include <cker/train/operation/BinaryArithmetic.h>
+#include <cker/train/operation/ReLU.h>
+
+namespace onert
+{
+namespace backend
+{
+namespace train
+{
+namespace ops
+{
+
+BinaryArithmeticLayer::BinaryArithmeticLayer()
+  : cpu::ops::BinaryArithmeticLayer(), _back_prop_lhs{nullptr}, _back_prop_rhs{nullptr},
+    _back_prop_output{nullptr}
+{
+  // DO NOTHING
+}
+
+void BinaryArithmeticLayer::configure(const IPortableTensor *lhs, const IPortableTensor *rhs,
+                                      IPortableTensor *output, IPortableTensor *back_prop_lhs,
+                                      IPortableTensor *back_prop_rhs,
+                                      const IPortableTensor *back_prop_output,
+                                      const ir::Activation activation,
+                                      const ArithmeticType arithmetic_type)
+{
+  if (arithmetic_type != ArithmeticType::kAdd)
+  {
+    throw std::runtime_error{"Unsupported binary operation"};
+  }
+
+  cpu::ops::BinaryArithmeticLayer::configure(
+    lhs, rhs, output, activation, static_cast<cpu::ops::ArithmeticType>(arithmetic_type));
+
+  _back_prop_lhs = back_prop_lhs;
+  _back_prop_rhs = back_prop_rhs;
+  _back_prop_output = back_prop_output;
+  _arithmetic_type = arithmetic_type;
+  _activation = activation;
+}
+
+void BinaryArithmeticLayer::forward(bool) { cpu::ops::BinaryArithmeticLayer::run(); }
+
+void BinaryArithmeticLayer::backward()
+{
+  // Calculate gradient for activation
+  const IPortableTensor *backprop_act;
+  switch (_activation)
+  {
+    case ir::Activation::NONE:
+      backprop_act = _back_prop_output;
+      break;
+    case ir::Activation::RELU:
+      nnfw::cker::train::ReLUGrad(getShape(_output), getBuffer<float>(_output),
+                                  getShape(_back_prop_output), getBuffer<float>(_back_prop_output),
+                                  getShape(_act_back_prop_output.get()),
+                                  getBuffer<float>(_act_back_prop_output.get()));
+      backprop_act = _act_back_prop_output.get();
+      break;
+    default:
+      throw std::runtime_error("train ConvolutionLayer: Unsupported activation type yet");
+  }
```

Comment:
just FYI. 

I recently updated this related part. (related PR : #12492 )
So, instead of directly calling activation grad(), you could use a function in OperationUtils.h

https://github.com/Samsung/ONE/blob/f444dab7e2de17b7cbcdff39e2d1bc647e3103bc/runtime/onert/backend/train/ops/OperationUtils.h#L39-L62
 
like here : 

https://github.com/Samsung/ONE/blob/f444dab7e2de17b7cbcdff39e2d1bc647e3103bc/runtime/onert/backend/train/ops/ConvolutionLayer.cc#L136-L146

Reply:
Thanks for letting me know :)
Updated.